### ISSUE TITLE
Item list normalizer that uses getResult() hydration

### ISSUE THREAD

**Issue by b-b3rn4rd** [kitpages/KitpagesDataGridBundle#41]
From time to time I keep some display logic inside my entities, since `StandardNormalizer` hydrates result to array, I have to duplicate logic written in entity or move it to the external file. Like in the example below. It would be handy to a have an option to hydrate result to entity and specify field as property or custom method.

```php
<?php
namespace Kimwild\CommonBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;

class Clients
{
    /* ... */
    public function getFullAddress()
    {
        $return = '';
        
        if ($this->getAddress()) {
            $return .= $this->getAddress();
        }
        
        if ($this->getSuburb()) {
            $return .= ' ' . $this->getSuburb() . ', ';
        }
        
        if ($this->getState()) {
            $return .= $this->getState();
        }
        
        if ($this->getPostcode()) {
            $return .= ' ' . $this->getPostcode();
        }
        
        if ($this->getCountry()) {
            $return .= ', ' . $this->getCountry();
        }
         return $return;
    } 
    /* ... */   
}
```

**Comment by philippe-levan** [comment]
Hum... There is one ugly way to solve it :
We can make the request twice... and inject the object result into the normalized result (in an specific key for example). This second request can be an optionnal request, only done if you need to access to the object.

Another solution whould be to check if it is possible to normalize the results from an object result (I believe if it is possible, the normalizer would be kind of awful).

Right now, I don't see any good solution. I will explore the first ugly solution. I believe it can be the best.

(note : if you find a good solution don't hesitate to tell me  :-) ).

Best regards,
Philppe

**Comment by b-b3rn4rd** [comment]
Thanks for the reply. I reckon that the second option is viable. At this stage I just duplicated my code, however, I believe that others might experience similar problem.
I will have a look how this can be archived this weekend.

Best Regards,
Bernard

**Comment by philippe-levan** [comment]
The PR #47 changes the way the query is hydrated. It allows to change the doctrine hydrator in order to provide both the flatten values and the object representation.

In the first version, only the flatten version is provided (just as now but without some remaining bugs). But we are a bit closer from a clean solution for your issue :-)

**Comment by eliecharra** [comment]
ping @b-b3rn4rd 
Is your problem solved with [v2.4.1](https://github.com/kitpages/KitpagesDataGridBundle/releases/tag/v2.4.1) ?

**Comment by b-b3rn4rd** [comment]
hi @eliecharra, yeah, it seems so. I didn't have a change to test it yet, since I've already finished the project. Many thanks for your contribution
